os/arch/xtensa/src/esp32/esp32_i2c.c

```diff
@@ -313,12 +320,24 @@ static int i2c_get_timeout(i2c_port_t i2c_num, int* timeout)
 	return 0;
 }
 
+static int i2c_esp32_check_bus(struct esp32_i2c_priv_s *priv)
```

Comment:
I think this is pretty risky.
You'd better  clock instead of loop count, because if there are many threads, running time will be increased.
Also naming should be changed, how about i2c_esp32_wait_until_ready or something.

Reply:
Hi, Mr. Kwon, This is interrupt handler, we could not use delay or sleep funcion, and also, there is no such multiple thread cases, because interrupt handler is unique, it will be called only when interrupt coming out.

Reply:
Got it.